### PR TITLE
chore: add guide for  using mermaid diagramming tool to docs

### PR DESCRIPTION
**Pull request by jencull**

### CODE REVIEW

File: docs/guide-diagrams.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+User guide for beginners
+https://mermaid.js.org/intro/n00b-gettingStarted.html
+
+Mermaid syntax cheat sheet 
+https://jojozhuang.github.io/tutorial/mermaid-cheat-sheet/
+
+Vscode extension Markdown Preview Mermaid Support
+https://marketplace.visualstudio.com/items?itemName=bierner.markdown-mermaid
+
+To view diagram in Vscode: right click on file and select ‘open preview’ after plugin has been installed (restart required)
```
Comment (dirgim) [inline docs/guide-diagrams.md:18]: Can you add some more viewing/developing information from [this comment](https://github.com/redhat-appstudio/integration-service/pull/219#issue-1757381642) somewhere around here?

**Comment by dirgim** [comment]
Could you also add a section explaining where the diagrams for the Integration service controllers are stored ([docs/](https://github.com/redhat-appstudio/integration-service/tree/main/docs)) and also explaining that we should submit updates to those diagrams alongside any changes to controller behavior?

**Comment by jencull** [comment]
> @jencull could you also add a section explaining where the diagrams for the Integration service controllers are stored ([docs/](https://github.com/redhat-appstudio/integration-service/tree/main/docs)) and also explaining that we should submit updates to those diagrams alongside any changes to controller behavior?

Done.

**Review by dirgim — APPROVED** [review]
LGTM

**Comment by MartinBasti** [comment]
Maybe it should be named `README.md` and also existing diagrams linked there, so it is shown nicely in GH web

**Comment by jencull** [comment]
https://github.com/redhat-appstudio/integration-service/blob/main/docs/snapshot-controller.md

> Maybe it should be named `README.md` and also existing diagrams linked there, so it is shown nicely in GH web

Done.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+## Creating or editing Mermaid diagrams
+
+Mermaid is a JS based diagramming tool that renders markdown style syntax to create/modify diagrams. Mermaid has [native support in Github](https://github.com/github/roadmap/issues/372)
+
+## Existing Diagrams
+- [binding-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/binding-controller.md)
+- [scenario-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/scenario-controller.md)
+- [snapshot -controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/snapshot-controller.md)
```
Comment (MartinBasti) [inline docs/README.md:9]:
```suggestion
- [snapshot-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/snapshot-controller.md)
```

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
+## Creating or editing Mermaid diagrams
+
+Mermaid is a JS based diagramming tool that renders markdown style syntax to create/modify diagrams. Mermaid has [native support in Github](https://github.com/github/roadmap/issues/372)
+
+## Existing Diagrams
```
Comment (MartinBasti) [inline docs/README.md:6]: I'd move it to top level, so people see diagrams first and then how to update them

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- [binding-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/binding-controller.md)
+- [scenario-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/scenario-controller.md)
+- [snapshot -controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/snapshot-controller.md)
+
+## Editing Diagrams
+- Diagrams are stored in the Docs folder in the repository. As changes are made to the behaviour of controllers these changes should also be updated in the diagram. The updated mermaid diagram doc should be submitted as part of the PR. 
+- To view diagram in Vscode: right click on file and select ‘open preview’ after plugin has been installed (restart required)
```
[the remark this answers is not in the document]
Comment (jencull) [reply docs/README.md]: done

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+Mermaid is a JS based diagramming tool that renders markdown style syntax to create/modify diagrams. Mermaid has [native support in Github](https://github.com/github/roadmap/issues/372)
+
+## Existing Diagrams
+- [binding-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/binding-controller.md)
+- [scenario-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/scenario-controller.md)
+- [snapshot -controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/snapshot-controller.md)
+
+## Editing Diagrams
+- Diagrams are stored in the Docs folder in the repository. As changes are made to the behaviour of controllers these changes should also be updated in the diagram. The updated mermaid diagram doc should be submitted as part of the PR.
```
[the remark this answers is not in the document]
Comment (jencull) [reply docs/README.md]: done

File: docs/README.md (hunk 3 above)
Comment (jencull) [reply docs/README.md]: done

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+- [scenario-controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/scenario-controller.md)
+- [snapshot -controller](https://github.com/redhat-appstudio/integration-service/blob/main/docs/snapshot-controller.md)
+
+## Creating or editing Mermaid diagrams
+
+Mermaid is a JS based diagramming tool that renders markdown style syntax to create/modify diagrams. Mermaid has [native support in Github](https://github.com/github/roadmap/issues/372)
+
+## Editing Diagrams
+- Diagrams are stored in repository's `/docs` folder in the repository. As changes are made to the behaviour of controllers these changes should also be updated in the diagram. 
+- The updated mermaid diagram can be submitted as part of a seperate PR or part of the PR that changes the behaviour of the controller..
```
Comment (dheerajodha) [inline docs/README.md:13]: nit:
```suggestion
- The updated mermaid diagram can be submitted as part of a separate PR or part of the PR that changes the behaviour of the controller.
```

Comment (jencull) [reply docs/README.md]: done

Comment (dheerajodha) [reply docs/README.md]: I can still see the typo: seperate -> separate

File: docs/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dheerajodha) [reply docs/README.md]: You forgot to delete the `in the repository` (and are also missing the comma) as suggested above.

File: docs/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dheerajodha) [reply docs/README.md]: There's no space after the word "plugin".
The hyphen and full stop are missing too but they're optional.

File: docs/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jencull) [reply docs/README.md]: done

File: docs/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jencull) [reply docs/README.md]: done

File: docs/README.md (hunk 6 above)
Comment (jencull) [reply docs/README.md]: done

File: docs/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dheerajodha) [reply docs/README.md]: thanks!

File: docs/README.md (hunk 6 above)
Comment (dheerajodha) [reply docs/README.md]: thanks!

**Review by MartinBasti — APPROVED** [review]

**Pull request merged**